libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1190,23 +1198,28 @@ void IRGeneratorForStatements::appendExternalFunctionCall(
 		encodeInPlace = false;
 	bool encodeForLibraryCall = funKind == FunctionType::Kind::DelegateCall;
 	solUnimplementedAssert(!encodeInPlace, "");
-	solUnimplementedAssert(!funType.padArguments(), "");
+	solUnimplementedAssert(funType.padArguments(), "");
 	templ("encodeArgs", abi.tupleEncoder(argumentTypes, funType.parameterTypes(), encodeForLibraryCall));
 	templ("argumentString", argumentString);
 
 	// Output data will replace input data, unless we have ECRecover (then, output
 	// area will be 32 bytes just before input area).
-	templ("retSize", to_string(retSize));
+	if (dynamicReturnSize)
+		templ("retSize", "returndatasize()");
```

Comment:
I think it would be good to have a different name for a variable whose value will only be valid after the call - `retSize` is also used before the call.